### ISSUE TITLE
Publishing error

### ISSUE THREAD

**Issue by miclefebvre** [GradleUp/shadow#306]
### Shadow Version
2.0.0

### Gradle Version
4.0

### Expected Behavior
publish to maven local

### Actual Behavior
> Failed to publish publication 'shadow' to repository 'mavenLocal'
   > Invalid publication 'shadow': POM file is invalid. Check any modifications you have made to the POM file.

It seems't that the generated publication pom file isn't working.

content of the publication pom:

<?xml version="1.0" encoding="UTF-8"?>
<project xmlns="http://maven.apache.org/POM/4.0.0" xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance">
  <modelVersion>4.0.0</modelVersion>
  <groupId>com.patate</groupId>
  <artifactId>patate-g</artifactId>
  <version>1.8.0-SNAPSHOT</version>
  <packaging>pom</packaging>
  <name>patate-g</name>
  <description/>
  <dependencies/>
  <dependencies/>
  <dependencies/>
</project>

line added to the gradle file:

publishing {
    publications {
        shadow(MavenPublication) { publication ->
            project.shadow.component(publication)
        }
    }
}

The shadowJar is working.

**Comment by DPUkyle** [comment]
I'm dealing with the same situation @miclefebvre. Is your project also using `java-gradle-plugin`?

**Comment by miclefebvre** [comment]
@DPUkyle yes also.

**Comment by DPUkyle** [comment]
OK good. I'm working on a fix or workaround.

The problem appears to be that `com.github.jengelman.gradle.plugins.shadow.ShadowExtension#component(MavenPublication)` doesn't play nice with the `java-gradle-plugin`, which has already created a `MavenPublication` called `pluginMaven`.

In simpler terms, the `component` method always adds a new `<dependencies>` node, but should instead first check for the existing one created by `java-gradle-plugin`. I hope to create a PR today... if I don't drive myself crazy trying to figure out xml parsing in groovy first.

**Comment by miclefebvre** [comment]
Amazing

**Comment by rgozim** [comment]
Any update on this? The issue still exists on version 2.0.4

**Comment by felipecsl** [comment]
make sure you're using the syntax below:
```
publishing {
  publications {
    shadow(MavenPublication) {
      project.shadow.component(it)
      artifact jar
    }
  }
}
```

**Comment by Goooler** [comment]
https://github.com/GradleUp/shadow/tree/main/src/docs/publishing
